catt/stream_info.py

```diff
@@ -148,9 +149,19 @@ def set_playlist_entry(self, number):
         else:
             raise StreamInfoError("called on non-playlist")
 
-    def _get_local_ip(self):
-        interface = netifaces.gateways()["default"][netifaces.AF_INET][1]
-        return netifaces.ifaddresses(interface)[netifaces.AF_INET][0]["addr"]
+    def _get_local_ip(self, host):
+        for adapter in ifaddr.get_adapters():
+            for aip in adapter.ips:
+                try:
+                    ipaddress.IPv4Address(aip.ip)
```

Comment:
This looks great, except is there a reason to not support IPv6 here?

Reply:
I figured it wouldn't be relevant on the users own network, but I can implement it if you prefer, no problem.

Reply:
I know a few people who run their internal network as IPv6-only, I just don't want to exclude them if it's easy to include them. Won't ifaddr return the IPv6 subnet in the same way as the IPv4 one? How much work is it to do this?

Reply:
> Won't ifaddr return the IPv6 subnet in the same way as the IPv4 one? How much work is it to do this?

It will, it's the validating part that needs to change a bit. I'll fix it.

Reply:
Not as easy as I thought, but should be dandy now.